hbase-server/src/main/java/org/apache/hadoop/hbase/rsgroup/RSGroupInfoManagerImpl.java

```diff
@@ -1064,19 +1065,34 @@ private boolean isTableInGroup(TableName tableName, String groupName,
     return rit;
   }
 
-  private Map<TableName, Map<ServerName, List<RegionInfo>>> getRSGroupAssignmentsByTable(
-      String groupName) throws IOException {
+  /**
+   * This is an EXPENSIVE clone.  Cloning though is the safest thing to do.
+   * Can't let out original since it can change and at least the load balancer
+   * wants to iterate this exported list.
```

Comment:
Maybe for these two lines, we can mention:
```
Load balancer should iterate over this list because cloned list will ignore disabled table and split parent region cases.
```

Reply:
I have added this comment